pkg/importer/dataStream_test.go

```diff
@@ -7,416 +7,404 @@ import (
-		urlPath = getURLPath(testfile)
-	}
-	testFilePath := getTestFilePath(testfile)
-	ds := createDataStream(urlPath, "", "")
+	It("create expected random name", func() {
+		randName := make([]byte, numbyte)
+		rand.Read(randName)
+		wantString := hex.EncodeToString(randName)
 
-	// if no file supplied just return empty byte stream
-	if len(testfile) == 0 && !defaultBuf {
-		return ds, nil, nil
-	}
-	if len(testfile) == 0 && defaultBuf {
-		return ds, []byte{'T', 'E', 'S', 'T'}, nil
-	}
+		got := randTmpName("testfile.img")
+		_, fn := filepath.Split(got)
 
-	f, _ := os.Open(testFilePath)
-	defer f.Close()
-	testBytes, err := ioutil.ReadAll(f)
-	if err != nil {
-		return nil, nil, fmt.Errorf("Unable to read datastream buffer")
-	}
+		Expect(len(fn)).To(Equal(len("testfile.img") + len(wantString)))
+	})
+})
 
-	return ds, testBytes, nil
-}
+var _ = Describe("Streaming Data Conversion", func() {
 
-func getFileSize(testfile string) (int, error) {
-	f, err := os.Open(getTestFilePath(testfile))
-	defer f.Close()
+	tinyCoreIso := "tinyCore.iso"
+	baseTestImage, err := filepath.Abs(filepath.Join(TestImagesDir, tinyCoreIso))
 	if err != nil {
-		return 0, fmt.Errorf("Unable to open source datastream file %s", getTestFilePath(testfile))
+		Fail(fmt.Sprintf("Error getting abs path: %v\n", err))
 	}
-	fi, err := f.Stat()
-	if err != nil {
-		return 0, fmt.Errorf("Unable to stat file %v", f.Name())
-	}
-
-	return int(fi.Size()), nil
-}
-
-func getTestFilePath(testfile string) string {
-	// CWD is set within go test to the directory of the test
-	// being executed, so using relative path
-	imageDir, _ := filepath.Abs(TestImagesDir)
-	return filepath.Join(imageDir, testfile)
-}
 
-func getURLPath(testfile string) string {
-	// CWD is set within go test to the directory of the test
-	// being executed, so using relative path
-	imageDir, _ := filepath.Abs(TestImagesDir)
-	return "file://" + imageDir + "/" + testfile
-}
+	var tmpTestDir string
+	BeforeEach(func() {
+		tmpTestDir = testDir(os.TempDir())
+		if err != nil {
+			Fail(fmt.Sprintf("[BeforeEach] Failed created test dir: %v\n", err))
+		}
+		By(fmt.Sprintf("[BeforeEach] Creating temporary dir %s", tmpTestDir))
+		syscall.Umask(0000)
+		err = os.Mkdir(tmpTestDir, 0777)
+		if err != nil {
+			Fail(fmt.Sprintf("Could not create tmp file: %v\n ", err))
+		}
+	})
 
-func startHTTPServer(port int, dir string) (*http.Server, error) {
-	server := &http.Server{
-		Addr:    fmt.Sprintf(":%d", port),
-		Handler: http.FileServer(http.Dir(dir)),
-	}
+	AfterEach(func() {
+		By(fmt.Sprintf("[AfterEach] Cleaning up temporary dir %s", tmpTestDir))
+		os.RemoveAll(tmpTestDir)
+	})
 
-	go func() {
-		if err := server.ListenAndServe(); err != nil {
-			// cannot panic, because this probably is an intentional close
-			log.Printf("Httpserver: ListenAndServe() error: %s", err)
+	table.DescribeTable("when data is in a supported file format", func(originalFile string, useVirtSize bool, expectFormats []string) {
```

Comment:
seems like this test case is more than a unit test of a single main func. Can size be tested w/o calling Copy or CopyImage? For all tar files and qcow2 files the answer is yes -- the ds.Size field should work. But for iso files (including .gz and .xz) the answer is no AFAIK.

Reply:
This particular test was in the datastream_ginko_test file originally, so I am guessing it used to be what you guys called a functional test, so yes it is more than a single function test.